yarn/src/main/scala/org/apache/spark/deploy/yarn/ClientArguments.scala

```diff
@@ -103,11 +104,15 @@ private[spark] class ClientArguments(args: Array[String], sparkConf: SparkConf)
           userClass = value
           args = tail
 
+        case ("--primaryResource") :: value :: tail =>
```

Comment:
I agree, since we don't ever set this to the main jar if we're not running python